### PR TITLE
feat(provider): implement info fields

### PR DESCRIPTION
**Pull request by troian**
Signed-off-by: Artur Troian <troian.ap@gmail.com>

### CODE REVIEW

File: testutil/provider.go [hunk 1]
```diff
@@ -13,5 +13,9 @@ func Provider(t testing.TB) ptypes.Provider {
 		Owner:      AccAddress(t).String(),
 		HostURI:    Hostname(t),
 		Attributes: Attributes(t),
+		Info: ptypes.ProviderInfo{
+			EMail:   "test@example.com",
+			Website: "example.com",
```
Comment (boz) [inline testutil/provider.go:18]: `ProviderHostname(t)`?

File: x/provider/types/types.go [hunk 2]
```diff
@@ -3,17 +3,26 @@ package types
 import (
 	"bytes"
 	"fmt"
+	"net/url"
+	"regexp"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+var emailRegex = regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
```
Comment (boz) [inline x/provider/types/types.go:12]: I'd almost rather have no validation.

File: x/provider/types/types.go [hunk 3]
```diff
@@ -46,3 +55,21 @@ func (p *Provider) Address() sdk.AccAddress {
 
 	return owner
 }
+
+func (m ProviderInfo) IsEmpty() bool {
+	return m.EMail == "" && m.Website == ""
+}
+
+func (m ProviderInfo) Validate() error {
+	// net.LookupMX ?
```
Comment (boz) [inline x/provider/types/types.go:64]: no, this has to be stateless & deterministic - can't make api calls and such.

**Review by boz — APPROVED** [review]
:+1: looks good, make sure it works with `tx validator create` and `tx validator update`.

**Review by hydrogen18 — APPROVED** [review]

**Pull request merged**